### PR TITLE
[MANUAL MIRROR] Moves gauze removal signals to gauze destroy, removing weird seep_gauze usages + misc improvements

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Mirror of https://github.com/Skyrat-SS13/Skyrat-tg/pull/24023

Title.

## How This Contributes To The Skyrat Roleplay Experience

Gauze removal was handled by using seep_gauze(9999) instead of just deleting it, which was really fucking weird??

This makes behavior more sensical and modular (you can remove gauze without using seep_gauze now).

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

image

</details>

## Changelog

:cl:
code: Gauze removal is now handled by the gauze's destroy instead of seep_gauze
/:cl:

### CODE REVIEW

**Comment by lessthnthree** [comment]
Do you want https://github.com/Skyrat-SS13/Skyrat-tg/pull/24023 closed?

**Comment by nikothedude** [comment]
WHY DID IT SQUASH THE FUCKING COMMITS GITHUB I FUCKING HATE YOU

**Comment by nikothedude** [comment]
> Do you want #24023 closed?

sure

**Comment by vinylspiders** [comment]
> WHY DID IT SQUASH THE FUCKING COMMITS GITHUB I FUCKING HATE YOU

At least this one is shorter than the last one...

For future reference, my process for making manual mirrors is using git bash:
If you haven't already, create a remote for tgstation on your repo.
`git remote add tgstation https://github.com/tgstation/tgstation`
Then whenever you need to update it to get new commits:
`git fetch tgstation` 
Then do a cherry pick:
`git cherry-pick -m 1 2bdadebbadab0de472efd9993d4b1595daee16e9` 
(that last arg is the merge commit for this mirror, but obviously you'd just swap that out with whatever one you want to cherry pick.

And then your commits should just be there, no squashing.

File: modular_skyrat/modules/medical/code/wounds/medical.dm [hunk 1]
```diff
@@ -14,25 +12,16 @@
...
 	. = ..()
 
-	UnregisterSignal(src, COMSIG_BODYPART_GAUZED)
+	owner?.update_bandage_overlays()
 
-/// Signal handler that allows us to modularly detect if we were applied to a limb or not.
-/obj/item/bodypart/proc/got_gauzed(datum/signal_source, obj/item/stack/medical/gauze/new_gauze)
-	SIGNAL_HANDLER
+/obj/item/stack/medical/gauze/Destroy()
+	var/mob/living/carbon/cached_owner = gauzed_bodypart?.owner
```
Comment (vinylspiders) [inline modular_skyrat/modules/medical/code/wounds/medical.dm]:
```suggestion
	var/mob/living/carbon/previously_gauzed = gauzed_bodypart?.owner
```
This is just a clarity thing, it's not actually being cached here this is just a local var that only exists in the frame of this proc. Caching implies it's committed to memory.

Just changing it to be less misleading.

File: modular_skyrat/modules/medical/code/wounds/medical.dm [hunk 2]
```diff
@@ -14,25 +12,16 @@
...
-		var/obj/item/stack/medical/gauze/applied_gauze = current_gauze
-		applied_gauze.set_limb(src) // new_gauze isnt actually the gauze that was applied weirdly
+	. = ..()
 
-/obj/item/stack/medical/gauze/Destroy()
-	bodypart?.current_gauze = null
-	bodypart?.owner?.update_bandage_overlays()
-	set_limb(null)
-	return ..()
+	cached_owner?.update_bandage_overlays()
```
Comment (vinylspiders) [inline modular_skyrat/modules/medical/code/wounds/medical.dm]:
```suggestion
	previously_gauzed?.update_bandage_overlays()
```

File: modular_skyrat/modules/medical/code/wounds/medical.dm [hunk 3]
```diff
@@ -95,36 +84,31 @@
...
 	if(href_list["remove"])
-		if(!bodypart.owner)
+		if(!gauzed_bodypart.owner)
 			return
 		if(!iscarbon(usr))
 			return
-		if(!in_range(usr, bodypart.owner))
+		if(!in_range(usr, gauzed_bodypart.owner))
 			return
 		var/mob/living/carbon/C = usr
```
Comment (vinylspiders) [inline modular_skyrat/modules/medical/code/wounds/medical.dm]:
```suggestion
		var/mob/living/carbon/carbon_user = usr
```
I know you didn't write this to begin with but just getting rid of the single letter vars while we're in here

File: modular_skyrat/modules/medical/code/wounds/medical.dm [hunk 4]
```diff
@@ -95,36 +84,31 @@
...
-		var/self = (C == bodypart.owner)
-		C.visible_message(span_notice("[C] begins removing [name] from [self ? "[bodypart.owner.p_Their()]" : "[bodypart.owner]'s" ] [bodypart.name]..."), span_notice("You begin to remove [name] from [self ? "your" : "[bodypart.owner]'s"] [bodypart.name]..."))
-		if(!do_after(C, (self ? SELF_AID_REMOVE_DELAY : OTHER_AID_REMOVE_DELAY), target=bodypart.owner))
+		var/self = (C == gauzed_bodypart.owner)
+		C.visible_message(span_notice("[C] begins removing [name] from [self ? "[gauzed_bodypart.owner.p_Their()]" : "[gauzed_bodypart.owner]'s" ] [gauzed_bodypart.name]..."), span_notice("You begin to remove [name] from [self ? "your" : "[gauzed_bodypart.owner]'s"] [gauzed_bodypart.name]..."))
+		if(!do_after(C, (self ? SELF_AID_REMOVE_DELAY : OTHER_AID_REMOVE_DELAY), target = gauzed_bodypart.owner))
```
Comment (vinylspiders) [inline modular_skyrat/modules/medical/code/wounds/medical.dm]:
```suggestion
		var/self = (carbon_user == gauzed_bodypart.owner)
		carbon_user.visible_message(span_notice("[carbon_user] begins removing [name] from [self ? "[gauzed_bodypart.owner.p_Their()]" : "[gauzed_bodypart.owner]'s" ] [gauzed_bodypart.name]..."), span_notice("You begin to remove [name] from [self ? "your" : "[gauzed_bodypart.owner]'s"] [gauzed_bodypart.name]..."))
		if(!do_after(carbon_user, (self ? SELF_AID_REMOVE_DELAY : OTHER_AID_REMOVE_DELAY), target = gauzed_bodypart.owner))
```

File: modular_skyrat/modules/medical/code/wounds/medical.dm [hunk 5]
```diff
@@ -95,36 +84,31 @@
...
 			return
 		if(QDELETED(src))
 			return
-		C.visible_message(span_notice("[C] removes [name] from [self ? "[bodypart.owner.p_Their()]" : "[bodypart.owner]'s" ] [bodypart.name]."), span_notice("You remove [name] from [self ? "your" : "[bodypart.owner]'s" ] [bodypart.name]."))
+		C.visible_message(span_notice("[C] removes [name] from [self ? "[gauzed_bodypart.owner.p_Their()]" : "[gauzed_bodypart.owner]'s" ] [gauzed_bodypart.name]."), span_notice("You remove [name] from [self ? "your" : "[gauzed_bodypart.owner]'s" ] [gauzed_bodypart.name]."))
```
Comment (vinylspiders) [inline modular_skyrat/modules/medical/code/wounds/medical.dm]:
```suggestion
		carbon_user.visible_message(span_notice("[carbon_user] removes [name] from [self ? "[gauzed_bodypart.owner.p_Their()]" : "[gauzed_bodypart.owner]'s" ] [gauzed_bodypart.name]."), span_notice("You remove [name] from [self ? "your" : "[gauzed_bodypart.owner]'s" ] [gauzed_bodypart.name]."))
```

File: modular_skyrat/modules/medical/code/wounds/medical.dm [hunk 6]
```diff
@@ -95,36 +84,31 @@
...
 		var/obj/item/gotten = rip_off()
 		if(gotten && !C.put_in_hands(gotten))
 			gotten.forceMove(get_turf(C))
```
Comment (vinylspiders) [inline modular_skyrat/modules/medical/code/wounds/medical.dm]:
```suggestion
		if(gotten && !carbon_user.put_in_hands(gotten))
			gotten.forceMove(get_turf(carbon_user))
```

**Review by vinylspiders — CHANGES_REQUESTED** [review]
Looks good to me otherwise!

**Review by vinylspiders — APPROVED** [review]

**Pull request merged**